Wrappers/Python/ccpi/viewer/CILViewerBase.py

```diff
@@ -0,0 +1,278 @@
+
+        #ViewerEvent
+        self.event = ViewerEventManager()
+
+        self.histogramPlotActor = vtk.vtkXYPlotActor()
+        self.histogramPlotActor.ExchangeAxesOff()
+        self.histogramPlotActor.SetXLabelFormat("%g")
+        self.histogramPlotActor.SetXLabelFormat("%g")
+        self.histogramPlotActor.SetAdjustXLabels(3)
+        self.histogramPlotActor.SetXTitle("Level")
+        self.histogramPlotActor.SetYTitle("N")
+        self.histogramPlotActor.SetXValuesToValue()
+        self.histogramPlotActor.SetPlotColor(0, (0, 1, 1))
+
+    def getInteractor(self):
+        return self.iren
+
+    def getRenderer(self):
+        return self.ren
+
+    def getRenderWindow(self):
+        '''returns the render window'''
+        return self.renWin
+
+    def getSliceOrientation(self):
+        return self.sliceOrientation
+
+    def setActiveSlice(self, sliceno):
+        self.slicenos[self.getSliceOrientation()] = sliceno
+
+    def getActiveSlice(self):
+        return self.slicenos[self.getSliceOrientation()]
+
+    def getSliceColorWindow(self):
+        return self.imageSlice.GetProperty().GetColorWindow()
+
+    def getSliceColorLevel(self):
+        return self.imageSlice.GetProperty().GetColorLevel()
+
+    def getSliceWindowLevelFromRange(self, cmin, cmax):
+        # set the level to the average between the percentiles
+        level = (cmin + cmax) / 2
+        # accommodates all values between the level an the percentiles
+        window = cmax - cmin
+
+        return window, level
+
+    def startRenderLoop(self):
+        self.iren.Start()
+
+    def getImageHistogramStatistics(self, method):
+        '''
+        returns histogram statistics for either the image
+        or gradient of the image depending on the method
+        '''
+        ia = vtk.vtkImageHistogramStatistics()
+        if method == 'scalar':
+            ia.SetInputData(self.img3D)
+        else:
+            grad = vtk.vtkImageGradientMagnitude()
+            grad.SetInputData(self.img3D)
+            grad.SetDimensionality(3)
+            grad.Update()
+            ia.SetInputData(grad.GetOutput())
+        ia.Update()
+        return ia
+
+    def getVolumeMapRange(self, percentiles, method):
+        '''
+        uses percentiles to generate min and max values in either
+        the image or image gradient (depending on method) for which
+        the colormap or opacity are displayed.
+        '''
+        ia = self.getImageHistogramStatistics(method)
+        ia.SetAutoRangePercentiles(*percentiles)
+        ia.Update()
+        min, max = ia.GetAutoRange()
+        return min, max
+
+    def getVolumeRange(self, method):
+        '''
+        Parameters
+        -----------
+        method: string : ['scalar', 'gradient']
+            'scalar' - returns full range of values in image
+            'gradient' - returns full range of values in image gradient
+        '''
+
+        return self.getVolumeMapRange((0, 100), method)
```

Comment:
I understand why you'd get the min and max from 0 and 100 percentiles, but wouldn't it be clearer to do 
```python

ia = self.getImageHistogramStatistics(method)
return ia.GetMinimum(), ia.GetMaximum()
```